File: services/src/test/java/com/senacor/code/fullstack/chat/service/ChannelServiceTest.java
package com.senacor.code.fullstack.chat.service;

import com.senacor.code.fullstack.chat.domain.Channel;
import com.senacor.code.fullstack.chat.repository.ChannelRepository;
import org.junit.Before;
import org.junit.Test;

import java.util.List;

import static java.util.Arrays.asList;
import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;


public class ChannelServiceTest {

    private ChannelRepository repository = mock(ChannelRepository.class);

    private ChannelService service = new ChannelService(repository);

    @Before
    public void setup() {
        when(repository.findAll()).thenReturn(asList(
                new Channel("general", "general"),
                new Channel("dev", "dev"),
                new Channel("humor", "humor")));
    }

    @Test
    public void fetchAllChannels() {
        List<Channel> expected = asList(
                new Channel("general", "general"),
                new Channel("dev", "dev"),
                new Channel("humor", "humor"));

        List<Channel> result = service.loadChannels();

        assertEquals(expected, result);
    }

    @Test
    public void existsChannel() {
        when(repository.existsById("dev")).thenReturn(true);

        boolean result = service.existsChannel("dev");

        assertTrue(result);
    }

    @Test
    public void existsChannelForNotExistingChannel() {
        when(repository.existsById("not-a-channel")).thenReturn(false);

        boolean result = service.existsChannel("not-a-channel");

        assertFalse(result);
    }

}

File: services/src/test/kotlin/com/senacor/code/fullstack/chat/channel/ChannelServiceTest.kt
package com.senacor.code.fullstack.chat.channel

import io.mockk.every
import io.mockk.mockk
import org.junit.Assert.*
import org.junit.Before
import org.junit.Test

class ChannelServiceTest {

    private val repository = mockk<ChannelRepository>()

    private val service = ChannelService(repository)

    private val expectedChannels = listOf(
        Channel("general", "general"),
        Channel("dev", "dev"),
        Channel("humor", "humor"))

    @Before
    fun setup() {
        every { repository.findAll() } returns expectedChannels
    }

    @Test
    fun fetchAllChannels() {
        val result = service.loadChannels()

        assertEquals(expectedChannels, result)
    }

    @Test
    fun existsChannel() {
        every { repository.existsById("dev") } returns true

        val result = service.existsChannel("dev")

        assertTrue(result)
    }

    @Test
    fun existsChannelForNotExistingChannel() {
        every { repository.existsById("not-a-channel") } returns false

        val result = service.existsChannel("not-a-channel")

        assertFalse(result)
    }

}